### PR TITLE
Add socket proxy to snap

### PR DESCRIPTION
**Pull request by CristianPrundeanu**
Socket proxy is part of the container signing feature.  It will proxy
the docker unix socket and eventually filter messages.

### CODE REVIEW

**Review by johnmichaelwelsh — APPROVED** [review]
Simple enough if you can get it to not error out.

**Comment by costanic** [comment]
Can we give this thing a more descriptive name than 'socket proxy'?

**Comment by moonchen** [comment]
> Can we give this thing a more descriptive name than 'socket proxy'?

Any ideas on what we should name it?

**Comment by costanic** [comment]
> > Can we give this thing a more descriptive name than 'socket proxy'?
> 
> Any ideas on what we should name it?

In our meeting we discussed docker-socket-proxy as a slightly less generic but still unimaginative name.  @johnmichaelwelsh threw out "TuProx".

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -637,3 +640,13 @@ parts:
...
+    socket-proxy:
+      plugin: cmake
+      source: https://github.com/PelionIoT/socket-proxy.git
+      source-type: git
+      source-tag: v1.0.2
+      override-build: |
+        snapcraftctl build
+        install ${SNAPCRAFT_PROJECT_DIR}/files/socket-proxy/scripts/launch-socket-proxy.sh ${SNAPCRAFT_PART_INSTALL}/
+        install -d ${SNAPCRAFT_PART_INSTALL}/wigwag/etc
+        git describe --tags > ${SNAPCRAFT_PART_INSTALL}/wigwag/etc/socket-proxy.VERSION
```
Comment (costanic) [inline snap/snapcraft.yaml:652]: Looks like the print-versions.sh script needs to be modified too: https://github.com/PelionIoT/snap-pelion-edge/blob/dev/files/version/print-versions.sh

**Comment by CristianPrundeanu** [comment]
Removed Do Not Merge based on Mo's comment about being code complete. Pending resolved PR comments of course.

File: snap/snapcraft.yaml (hunk 1 above)
Comment (johnmichaelwelsh) [reply snap/snapcraft.yaml]: done.

**Review by johnmichaelwelsh — APPROVED** [review]

**Pull request merged**